ledger/store/store.go

```diff
@@ -38,30 +38,28 @@ import (
 // this solution allows us to store payloads on disk with a negligible impact to insertion performance and a limited
 // impact to memory usage.
 
-// persistInterval is the time interval at which the store evicts the oldest elements from its LRU cache and stores
-// them persistently in the on-disk database.
-const persistInterval = 100 * time.Millisecond
-
 // Store is a component that provides fast persistent storage by using an LRU cache from which evicted entries get
 // persisted on disk.
 type Store struct {
 	log zerolog.Logger
 
-	db    *badger.DB
-	sema  *semaphore.Weighted
-	tx    *badger.Txn
-	mutex *sync.RWMutex   // guards the current transaction against concurrent access
-	wg    *sync.WaitGroup // keeps track of when the flush goroutine should exit
-	err   chan error
+	db   *badger.DB
+	sema *semaphore.Weighted
+	tx   *badger.Txn
+	txMu *sync.RWMutex   // guards the current transaction against concurrent access
+	wg   *sync.WaitGroup // keeps track of when the flush goroutine should exit
+	err  chan error
 
 	cache     *lru.Cache
 	cacheSize int
+	isDirtyMu *sync.RWMutex      // guards the dirty tracking map against concurrent access
+	isDirty   map[hash.Hash]bool // keeps track of whether cached entries are on disk
```

Comment:
This is OK, I guess, but I think a more memory-efficient way would be to add the dirty flag natively to the items in the cache.

Reply:
It's true that now that we have control over the cache layer, we can do that. I'll do it real quick, that should indeed improve clarity and makes a lot of sense.